client/src/main.tsx

```diff
@@ -1,24 +1,21 @@
+import React from "react";
 import ReactDOM from "react-dom/client";
 import { BrowserRouter } from "react-router-dom";
+import { initializeApp } from "firebase/app";
+
 import App from "./App.tsx";
 import "./index.css";
-
-import { initializeApp } from "firebase/app";
-import React from "react";
-import { AuthProvider } from "./contexts/AuthContext.tsx";
-import { TripProvider } from "./contexts/TripContext.tsx";
+import { ContextProvider } from "./contexts/Context.tsx";
 
 const config = JSON.parse(import.meta.env.VITE_FIREBASE_CONFIG);
 const app = initializeApp(config);
 
 ReactDOM.createRoot(document.getElementById("root")!).render(
   <React.StrictMode>
     <BrowserRouter>
-      <AuthProvider>
-        <TripProvider>
-          <App />
-        </TripProvider>
-      </AuthProvider>
+      <ContextProvider>
+        <App />
+      </ContextProvider>
```

Comment:
new `<ContextProvider />` component removes the need to keep nesting our context providers. instead, see `contexts/Context.tsx`